pallets/eth-bridge/src/lib.rs

```diff
@@ -514,22 +523,30 @@ pub mod pallet {
         ) -> DispatchResultWithPostInfo {
             ensure_none(origin)?;
 
+            ensure!(events_fraction.is_valid(), Error::<T>::InvalidEventVote);
             // Validate that the range is the current one.
             let active_range =
                 Self::active_ethereum_range().ok_or(Error::<T>::EventScanningDisabled)?;
-            ensure!(range == active_range, Error::<T>::EventScanningDisabled);
+            ensure!(range == active_range.range, Error::<T>::NonActiveEthereumRange);
+            ensure!(
+                active_range.status == EventProcessingStatus::UnderValidation,
+                Error::<T>::VotingEnded
+            );
 
             // Check if already vote has been casted
-            let votes = DiscoveredEventsShard::<T>::get(&range, &events_fraction);
+            let votes = DiscoveredEventsFractions::<T>::get(&range, &events_fraction);
```

Comment:
Personally I would rename `DiscoveredEventsFractions` to `EthereumEvents`

Reply:
yes, fraction has the sense of division, of quotient to me. A part of a whole that will not work until all fractions are brought together. But each event is valid on its own, isn't it? Perhaps this is just a collection a events, and there will be many of that type and size. I like `batch` for that, but lots of words would work: set, lot, group, etc.
You changed `shard` and maybe you had a reason for it. It is a loaded term in computer science, and may imply a number of meanings we don't want, so I'm ok with changing from it.